### PR TITLE
Add incomplete attributions to table

### PR DESCRIPTION
**Pull request by someshkhandelia**
### Summary of changes

Add incomplete attributions to table

### Context and reason for change

- We also fixed the PieChart Legend font-size.

Fix: #1126

Signed-off-by: someshkhandelia <someshkhandelia@gmail.com>

### CODE REVIEW

**Comment by someshkhandelia** [comment]
Previously,

image

Now,

image

File: src/Frontend/Components/ProjectStatisticsPopup/__tests__/ProjectStatisticsPopup.test.tsx [hunk 1]
```diff
@@ -117,9 +117,8 @@ describe('The ProjectStatisticsPopup', () => {
       screen.queryByText('Most Frequent Licenses')
     ).not.toBeInTheDocument();
     expect(screen.queryByText('Critical Signals')).not.toBeInTheDocument();
-    expect(
-      screen.queryByText('Incomplete Attributions')
-    ).not.toBeInTheDocument();
+    expect(screen.getByText('Incomplete Attributions')).toBeInTheDocument();
+    expect(screen.getAllByText('Incomplete Attributions')).not.toHaveLength(2);
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectStatisticsPopup/__tests__/ProjectStatisticsPopup.test.tsx:121]: here all positve numbers except 2 are ok: 1,3,4 would be good to assert more precisely. 

As above the assertion relies on an accidental coincidence

File: src/Frontend/Components/ProjectStatisticsPopup/__tests__/ProjectStatisticsPopup.test.tsx [hunk 2]
```diff
@@ -153,9 +152,8 @@ describe('The ProjectStatisticsPopup', () => {
     renderComponentWithStore(<ProjectStatisticsPopup />, { store });
     expect(screen.getByText('Most Frequent Licenses')).toBeInTheDocument();
     expect(screen.getByText('Critical Signals')).toBeInTheDocument();
-    expect(
-      screen.queryByText('Incomplete Attributions')
-    ).not.toBeInTheDocument();
+    expect(screen.getByText('Incomplete Attributions')).toBeInTheDocument();
+    expect(screen.getAllByText('Incomplete Attributions')).not.toHaveLength(2);
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectStatisticsPopup/__tests__/ProjectStatisticsPopup.test.tsx:156]: as above

File: src/Frontend/Components/ProjectStatisticsPopup/project-statistics-popup-helpers.ts [hunk 3]
```diff
@@ -236,6 +237,9 @@ export function aggregateAttributionPropertiesFromAttributions(
     }
   }
 
+  attributionPropertyCounts['incomplete'] = Object.keys(
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectStatisticsPopup/project-statistics-popup-helpers.ts:240]: The entire function  `aggregateAttributionPropertiesFromAttributions` could be simplified by using the same logic as in `AttributionCountsPanel`.

**Review by benedikt-richter — COMMENTED** [review]
I have the feeling the tests are very fragile and rely on a coincidence. It was not introduced here (even though it's increased here). Consider improving on the tests e.g. by using aria labels for components and then to assert their presence.
There is a refactoring that could be done.

**Answered by @someshkhandelia**: You are perhaps correct @benedikt-richter. I think though that refactoring the testing style is out of scope of this issue. A more relevant future issue where this could be addressed: https://github.com/opossum-tool/OpossumUI/issues/1079

**Comment by benedikt-richter** [comment]
The title of this table should be improved. Seems not to match what is displayed. Maybe just take "Attributions"

File: src/Frontend/Components/ProjectStatisticsPopup/__tests__/ProjectStatisticsPopup.test.tsx (hunk 1 above)
Comment (someshkhandelia) [reply src/Frontend/Components/ProjectStatisticsPopup/__tests__/ProjectStatisticsPopup.test.tsx:121]: I tried to keep it exact as:
`expect(screen.getAllByText('Incomplete Attributions')).toHaveLength(1);`
 but the pre-commit hooks changed my code to just:
`expect(screen.getByText('Incomplete Attributions')).toBeInTheDocument();`
which was not conveying that the PieChart isn't there but the table is there.
That is why I came up with:
`expect(screen.getByText('Incomplete Attributions')).toBeInTheDocument();`
`expect(screen.getAllByText('Incomplete Attributions')).not.toHaveLength(2);`
This shows that although 'Incomplete Attributions' is present in the page, it is not present twice (the PieChart is not rendered).

File: src/Frontend/Components/ProjectStatisticsPopup/__tests__/ProjectStatisticsPopup.test.tsx (hunk 2 above)
Comment (someshkhandelia) [reply src/Frontend/Components/ProjectStatisticsPopup/__tests__/ProjectStatisticsPopup.test.tsx:156]: Please see: https://github.com/opossum-tool/OpossumUI/pull/1149#discussion_r1018973817

File: src/Frontend/Components/ProjectStatisticsPopup/__tests__/ProjectStatisticsPopup.test.tsx [hunk 4]
```diff
@@ -98,7 +98,7 @@ describe('The ProjectStatisticsPopup', () => {
     renderComponentWithStore(<ProjectStatisticsPopup />, { store });
     expect(screen.getByText('Most Frequent Licenses')).toBeInTheDocument();
     expect(screen.getByText('Critical Signals')).toBeInTheDocument();
-    expect(screen.getByText('Incomplete Attributions')).toBeInTheDocument();
+    expect(screen.getAllByText('Incomplete Attributions')).toHaveLength(2);
```
[the remark this answers is not in the document]
Comment (someshkhandelia) [reply src/Frontend/Components/ProjectStatisticsPopup/__tests__/ProjectStatisticsPopup.test.tsx:101]: Coincidentally, both of them have same names. If they didn't e.g. one was "Incomplete Attributions" and another was "Incomplete attributions", all the tests would look nicer, but to have a consistent and conventional UX, I did this.

File: src/Frontend/Components/ProjectStatisticsPopup/project-statistics-popup-helpers.ts (hunk 3 above)
Comment (someshkhandelia) [reply src/Frontend/Components/ProjectStatisticsPopup/project-statistics-popup-helpers.ts]: Done.

**Comment by someshkhandelia** [comment]
Previously,

image

Now,

image

File: src/Frontend/Components/PieChart/PieChart.tsx [hunk 5]
```diff
@@ -49,6 +49,10 @@ const classes = {
     color: OpossumColors.white,
     fontFamily: 'sans-serif',
   },
+  legendWrapperStyle: {
+    fontFamily: 'sans-serif',
+    fontSize: '12px',
```
Comment (benedikt-richter) [inline src/Frontend/Components/PieChart/PieChart.tsx:54]: 👍

**Review by benedikt-richter — APPROVED** [review]

File: src/Frontend/Components/PieChart/PieChart.tsx (hunk 5 above)
Comment (someshkhandelia) [reply src/Frontend/Components/PieChart/PieChart.tsx:54]: Thank you.

**Review by leslielazzarino — APPROVED** [review]

**Pull request merged**